README.md

```diff
@@ -0,0 +1,400 @@
+
+- This proposal allows for some non-sensical `CoerceUnsized` implementations resulting in odd unsizing coercions, though the restrictions on the `CoerceUnsized` trait try to limit them (for the time being).
+- Unsizing coercions are now able to run arbitrary user code, placing it into a similar category to `Deref` in that regard, effectively adding yet more user facing \*magic\* to the language.
+- The `Unsize` trait now depends on the `Pointee` trait which means any push for stabilization will depend on the stabilization of said trait.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+- As was discussed in the custom reborrows issue, we could make `CoerceUnsized` represent a more general user controlled `Coerce` mechanism.
+- This proposal is forwards compatible with exposing more dynamic unsizing behavior in the future, where for example the metadata is read from a field of the source type. To support that, a new trait `DynamicUnsize` could be introduced as the supertrait of `Unsize`, exposing the needed functions to extract the metadata. Then a blanket impl can be provided that implements `DynamicUnsize` for anything implementing `Unsize` with delegating the metadata extraction functions to the `Unsize` impl. The reason for why such a split would be necessary is that not all coercions can read from the source object (raw pointer unsizing for example), so there needs to be a way to differentiate on the trait bounds for the corresponding `CoerceUnsized` implementations.
+
+# Prior art
+[prior-art]: #prior-art
+
+There is another Pre-RFC that tries to improve Unsizing which can be found [here](https://internals.rust-lang.org/t/pre-rfc-improved-unsizing/16861).
+It does so by just allowing more impls of the current traits, while restricting them by taking visibilities of fields into account which complicates the traits in a (subjectively to the author) confusing way.
+And while the RFC makes the traits more flexible, it does not offer the same flexibility that this proposal offers.
+
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- Given the pin unsoundness proposal, assuming negative reason was a thing, would an impl permitting to go from `Pin<impl Unpin>` to `Pin<impl !Unpin>` be safe?
+- The compiler emitted implementations for the unsize trait, in particular the `Foo<..., T, ...>` case may collide with user implementations. Is this problematic? Should they be overridable?
+- Will this design prevent any scenarios from being ever supported?
+- As usual, naming. Given we might want to introduce multiple unsize traits for certain requirements, should the proposed trait stick to `Unsize` or something more specific like `FromMetadataUnsize`?
+
+# Future possibilities
+[future-possibilities]: #future-possibilities
+
+- Expand the compiler emitted implementations of `CoerceUnsized` to enums, such as `Option<T>: CoerceUnsized<Option<U>>` where `T: CoerceUnsized<U>`.
```

Comment:
Why won't this be possible with this RFC? Didn't you even have an example with `Option`? And why should they be compiler emitted and not in the standard library?

Reply:
The section doesnt say that its not possible with this RFC, its just something one could consider doing in the future but the RFC is not about that part. Whether  they are emitted by compiler or added manually doesn't matter either. its a future possibility in that these are things one could think aobut after the RFC got accepted